### PR TITLE
Sets HostNetwork to False for tests which do not require it

### LINKED ISSUE #69559: Some tests do not require HostNetwork=true in order to pass
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug
/sig testing

**What happened**:

Some of the tests are spawning pods with HostNetwork=true, but it is not required in order for them to pass. Some of the tests are:

    [sig-network] Networking Granular Checks: Pods should function for intra-pod communication: http [NodeConformance]
    [sig-network] Networking Granular Checks: Pods should function for intra-pod communication: udp [NodeConformance]
    [sig-network] Networking Granular Checks: Pods should function for node-pod communication: http [NodeConformance]
    [sig-network] Networking Granular Checks: Pods should function for node-pod communication: udp [NodeConformance]
    [sig-network] Networking Granular Checks: Services [Slow] should function for pod-Service: http
    [sig-network] Networking Granular Checks: Services [Slow] should function for pod-Service: udp
    [sig-network] Networking Granular Checks: Services [Slow] should function for node-Service: http
    [sig-network] Networking Granular Checks: Services [Slow] should function for node-Service: udp
    [sig-network] Networking Granular Checks: Services [Slow] should update endpoints: http
    [sig-network] Networking Granular Checks: Services [Slow] should update endpoints: udp

Additionally, pods with ``HostNetwork=true`` are not supported on Windows nodes. Running the tests using pods ``HostsNetwork=false`` will allow the tests to run properly on Windows nodes as well.

**What you expected to happen**:

Tests should also be able to run and pass using Windows nodes.

**How to reproduce it (as minimally and precisely as possible)**:

Run the mentioned tests using Windows nodes.

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): v1.12.0-rc.2
- Cloud provider or hardware configuration: N/A
- OS (e.g. from /etc/os-release): Windows Server 1803
- Kernel (e.g. `uname -a`): N/A
- Install tools: N/A
- Others:

### PR DESCRIPTION
**Pull request by claudiubelu**
**What this PR does / why we need it**:

Some tests are setting HostNetwork=true, even if it is not required
for them to pass.

This patch will set the HostNetwork to false for those tests, allowing
them to be run on Windows nodes as well.

/sig testing

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #69559

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by PatrickLang** [comment]
/milestone 1.13
We need this to fix tests running on Windows so we can enable the periodic test runs

**Review by feiskyer — COMMENTED** [review]
/lgtm

File: test/e2e/framework/util.go
Comment (caseydavenport) [inline test/e2e/framework/util.go]: Might want to rename this to `NewExecPodSpec`, since it is no longer strictly host networked.

File: test/e2e/framework/util.go
Comment (caseydavenport) [inline test/e2e/framework/util.go]: Will this function work if not in hostNetwork?

It seems to me that it is relying on hitting the host at a given port via localhost, which won't resolve the same when run in hostNetwork vs pod networked.

Comment (bclau) [reply test/e2e/framework/util.go]: Apparently so. The only place this is used in, is in the test ``should preserve source pod IP for traffic thru service cluster IP``, which is even marked as passed in the ``pull-kubernetes-e2e-kops-aws`` and the ``pull-kubernetes-e2e-gce`` jobs.

File: test/e2e/framework/util.go
Comment (bclau) [reply test/e2e/framework/util.go]: Done

**Comment by PatrickLang** [comment]
/LGTM
This fixes these conformance tests on Windows :)

**Comment by spiffxp** [comment]
/sig architecture
/hold
This LGTM but I would like to get a sanity check from someone else from sig-arch since this is conformance-related

**Comment by dims** [comment]
/lgtm

Thanks @bclau

**Review by timothysc — APPROVED** [review]
I like it! 

/lgtm

File: test/e2e/framework/util.go
Comment (bgrant0607) [reply test/e2e/framework/util.go]: Something is fishy about that. This is exec'ing into a pod and curling a localhost URL of a sleep pod.

https://github.com/kubernetes/kubernetes/blob/8616687c908d03f5bfdbb7cef03468d232c00132/test/e2e/framework/util.go#L3431
https://github.com/kubernetes/kubernetes/blob/7f23a743e8c23ac6489340bbb34fa6f1d392db9d/test/e2e/network/networking.go#L87
https://github.com/kubernetes/kubernetes/blob/ef195428a0c5b16e3f3621b7fe59fa151ca28929/cmd/kube-proxy/app/server.go#L507

**Comment by bgrant0607** [comment]
@thockin @caseydavenport @bowei: This needs a thorough review by someone from SIG Network.

Which tests are intending to test service reachability from a pod perspective, and which ones, if any, are deliberately testing host reachability?

**Comment by bgrant0607** [comment]
I removed lgtm and approval until we hear from SIG Network.

File: test/e2e/network/networking.go
Comment (thockin) [inline test/e2e/network/networking.go]: This too - DialFromNode should not work without HostNetwork

File: test/e2e/network/service.go
Comment (thockin) [inline test/e2e/network/service.go]: How would this work without hostNet?  It expects `ss` to be able to run and see kube-proxy

**Review by thockin — COMMENTED** [review]
I think a lot of these cannot be changed as you propose, but I also think a lot of the kube-proxy details should not be conformance tests.

**Comment by PatrickLang** [comment]
This needs to move to v1.14 so it can be updated once we have a decision on Windows conformance profiles and how they will be implemented.

**Comment by nikopen** [comment]
@PatrickLang any chance this is merged in the following 2 weeks? 
Code Freeze is in effect from next Friday.

**Comment by bclau** [comment]
Closed in favor of: https://github.com/kubernetes/kubernetes/pull/71468

**Comment by bclau** [comment]
For some reason, I cannot reopen this PR, but i've sent another PR for it. https://github.com/kubernetes/kubernetes/pull/78731

Additional information:

Running the tests affected by this PR on Linux nodes on different environments, yielded the following results:

-    azure-vnet-cni: https://paste.ubuntu.com/p/gYxsvWVqDc/
-    flannel l2bridge: https://paste.ubuntu.com/p/55YDFFHZzR/
-    flannel overlay: https://paste.ubuntu.com/p/mtrnP5JyxF/

**Pull request closed without merging**
